flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/CatalogManager.java

```diff
@@ -187,8 +187,8 @@ public void registerCatalog(String catalogName, Catalog catalog) {
 			throw new CatalogException(format("Catalog %s already exists.", catalogName));
 		}
 
-		catalogs.put(catalogName, catalog);
 		catalog.open();
+		catalogs.put(catalogName, catalog);
```

Comment:
Can you add a case for this?
You can create a Catalog extends `GenericInMemoryCatalog` and implement `open` to throw exception. The check the TableEnv should not contains this catalog.

Reply:
thanks for your suggestions.i'll add a test case.